### PR TITLE
Incentive module spec

### PR DESCRIPTION
**Pull request by jelysn**

### CODE REVIEW

File: x/incentive/spec/03_keeper.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+-->
+
+# Keepers
+
+## Reward pool address for LPs
+
+There's constant reward pool address per liquidity pool id which will be distributed per epoch.
+
+```go
+    func GetLPRewardsPoolAddress(poolId uint64) {
```
Comment (kenta-elys) [inline x/incentive/spec/03_keeper.md:12]: This function should be in Liquidity pool keeper. right?

File: x/incentive/spec/03_keeper.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        return authtypes.NewModuleAddress(fmt.Sprintf("lp_rewards_pool_%d", poolId))
+    }
+```
+
+DEX revenue and margin revenue to be distributed to specific pool liquidity providers is sent to the address and incentive module distribute the those tokens to relevant pool LPs.
+
+## Elys token to USDC conversion
```
Comment (kenta-elys) [inline x/incentive/spec/03_keeper.md:19]: Gas fees will be Elys and we will need to convert it to USDC before distributing. I think we have agreed on the fact that all the DEX revenues will be collected to the reward wallet in the form of USDC. Meaning inside incentive module, we don't need to do any conversions on the rewards collected as it is already USDC. So I think we will just need to distribute them appropriately.

File: x/incentive/spec/03_keeper.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+Reward distribution for LPs will need to iterate LP reward pools with positive balance.
+And configure total rewards for LPs for pool.
+
+```go
+Rewards = EpochTotalRewards * UserLPCommitment / TotalLPCommitment
+```
+
+Rewards for LPs should be combined with Eden, Eden boost and DEX/margin rewards.
+
+TODO: Problem in this logic - Users can run a bot to instantly put liquidity just before reward distribution and take it out. This happens because of zero lockup time and epoch based reward distribution. One of lockup or epoch based distribution modified. This is common problem on reward distribution per epoch.
```
Comment (kenta-elys) [inline x/incentive/spec/03_keeper.md:44]: Good catch! We will need to have a solution to this.

File: x/incentive/spec/03_keeper.md (hunk 1 above)
Comment (jelysn) [reply x/incentive/spec/03_keeper.md:12]: This would be good to put on incentive types folder - since this is related to rewards and this will need to be used on margin as well not just on amm module.

File: x/incentive/spec/03_keeper.md (hunk 3 above)
Comment (jelysn) [reply x/incentive/spec/03_keeper.md:44]: Yeah - will need input from @DreamProCatcher @cosmic-vagabond

File: x/incentive/spec/03_keeper.md (hunk 1 above)
Comment (kenta-elys) [reply x/incentive/spec/03_keeper.md:12]: How can he know unless access to liquidity pool module?

File: x/incentive/spec/03_keeper.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+Unless there's a way to track if the commitment was done past epoch and not unbonded so far.
+
+## Reward distribution logic for stakers
+
+This mechanism should be similar to Cosmos SDK's distribution module does except for epoch based operation.
```
[the remark this answers is not in the document]
Comment (kenta-elys) [reply x/incentive/spec/03_keeper.md:49]: Okay np.

File: x/incentive/spec/02_state.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        [ (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int", (gogoproto.nullable) = false, (gogoproto.moretags) = "yaml:\"amount\""];
+    // epoch identifier
+    string epoch_identifier = 2 [(gogoproto.moretags) = "yaml:\"epoch_identifier\""];
+    // start_time of the distribution
+    google.protobuf.Timestamp start_time = 3
+        [(gogoproto.stdtime) = true, (gogoproto.nullable) = false, (gogoproto.moretags) = "yaml:\"start_time\""];
+    // distribution duration
+    int64 num_epochs = 4 [(gogoproto.moretags) = "yaml:\"num_epochs\""];
+    int64 current_epoch = 5 [(gogoproto.moretags) = "yaml:\"current_epoch\""];
+    int64 eden_boost_apr = 6 [(gogoproto.moretags) = "yaml:\"eden_boost_apr\""];
```
Comment (cosmic-vagabond) [inline x/incentive/spec/02_state.md:38]: @jelysn same here

**Review by cosmic-vagabond — CHANGES_REQUESTED** [review]

File: x/incentive/spec/02_state.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+# State
+
+## Params
+
+```proto
+message Params {
+  option (gogoproto.goproto_stringer) = false;
+
+  repeated IncentiveInfo lp_incentives = 1 [(gogoproto.nullable) = false, (gogoproto.moretags) = "yaml:\"lp_incentives\""];
+  repeated IncentiveInfo stake_incentives = 2 [(gogoproto.nullable) = false, (gogoproto.moretags) = "yaml:\"stake_incentives\""];
```
[the remark this answers is not in the document]
Comment (jelysn) [reply x/incentive/spec/02_state.md:14]: That would be nice, this is just copy from the incentive module at the time of copy.

**Review by cosmic-vagabond — APPROVED** [review]
LGTM
